### DISCUSSION TITLE
Interactive cd using Neotree

### DISCUSSION THREAD

**Discussion by Maritsu** [nvim-neo-tree/neo-tree.nvim#747] [Q&A]
I was thinking about adding a button to change directories from my Neovim dashboard by selecting a directory with some file viewer and immediately thought of Neotree (as it's the tree plugin I use). The question is - **would it be possible to do something like an interactive directory selector**? i.e. a Neotree window pops up with a filesystem view (although only folders) and returns a directory the user selects.

**Comment by miversen33** [comment]
So the basic tools that are needed for something like this exist in Neo-tree's [filesystem](https://github.com/nvim-neo-tree/neo-tree.nvim/tree/v2.x/lua/neo-tree/sources/filesystem) code, however I don't think achieving what is being asked is feasible without quite a bit of work within Neo-tree unfortunately. Its a cool idea though! But I don't see something like this being supported by Neo-tree core at this time. Of course I am not the end all decision maker, if you think this is a feature you would want to see, you can always add a feature request and see what the other maintainers/contributors think.
